yas3fs/__init__.py

```diff
@@ -1713,21 +1723,19 @@ def getattr(self, path, fh=None):
         if self.cache.is_deleting(path):
             logger.debug("getattr path '%s' is deleting -- throwing ENOENT" % (path))
             raise FuseOSError(errno.ENOENT)
-
         with self.cache.get_lock(path): # To avoid consistency issues, e.g. with a concurrent purge
-            cache = True
             recheck_s3 = False
             if self.cache.is_empty(path):
+                cache = True
```

Comment:
Odd... is the `cache` local even used? (Or `recheck_s3`?)

Reply:
Honestly..it doesn't seem necessary to move `cache = True` to here, so I can't quite remember why I did. I'll try reverting it again.  I did notice `recheck_s3` doesn't seem to be used, I wasn't sure if there was something I was missing